app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/SessionsRepositoryTest.kt

```diff
@@ -0,0 +1,279 @@
+package io.github.droidkaigi.confsched2017.repository.sessions
+
+import com.sys1yagi.kmockito.invoked
+import com.sys1yagi.kmockito.mock
+import com.sys1yagi.kmockito.verify
+import io.github.droidkaigi.confsched2017.api.DroidKaigiClient
+import io.github.droidkaigi.confsched2017.model.OrmaDatabase
+import io.github.droidkaigi.confsched2017.model.Session
+import io.reactivex.Completable
+import io.reactivex.Single
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class SessionsRepositoryTest {
+
+    @Test
+    fun hasCacheSessions() {
+        // false
+        run {
+            val repository = SessionsRepository(
+                    SessionsLocalDataSource(mock()),
+                    SessionsRemoteDataSource(mock())
+            )
+
+            assertThat(repository.hasCacheSessions()).isFalse()
+        }
+
+        // true
+        run {
+            val repository = SessionsRepository(
+                    SessionsLocalDataSource(mock()),
+                    SessionsRemoteDataSource(mock())
+            )
+            repository.cachedSessions = mapOf(0 to Session())
+            repository.setIdDirty(false)
+
+            assertThat(repository.hasCacheSessions()).isTrue()
+        }
```

Comment:
Could you please add a case that `repository has any cached session, but repository is dirty`?